### DISCUSSION TITLE
React table generics... what happened?

### DISCUSSION THREAD

**Discussion by colindjk** [TanStack/table#4239] [General]
I was following development of the latest version of react table and was very excited about many aspects of the project. 

One of which was the flexibility of the typing that react-table offered. 

Currently I'm blocked on upgrading to ~`8.0.0`~ `8.*` from the beta. I can get that the current version still _can_ do everything the original did with some conditional typing, but I was wondering if there's a chance that some form of the original concept would be available to use (the `TableGenerics` pattern). 

Mainly curious - what are the plans around the type system currently? 

For the time being I'm sticking to the beta version as it's generally best suited for the use case I'm dealing with, but if redefining type modules is the go to then I'll get to some refactoring. 

And again, this is one gripe, overall the release for 8.0.0 has been amazingly helpful for development, thanks for the great work :slightly_smiling_face:

**Comment by KevinVandy** [comment]
Don't upgrade to 8.0, upgrade to 8.5

**Reply by tannerlinsley** [reply]
Yep, 8.x ditched generic maps due to user-bound complexity (even though they were nice internally). Things are just as good now, maybe even better as we've moved to only 2 optional slotted generics for row data and column value and the rest of the generics have been moved into declaration merge-able type interfaces.

**Reply by LinnJS** [reply]
Thank you for all the rockin' work you all are doing!
